apps/snitch_core/lib/core/domain/order/transitions.ex

```diff
@@ -194,4 +196,118 @@ defmodule Snitch.Domain.Order.Transitions do
         {:halt, error}
     end)
   end
+
+  # Guard Class to check negative amount is skipped.
+
+  defp calc_payable_amount(order) do
+    packages_total = Package.compute_packages_total(order)
+    total_amount = Money.add!(packages_total, order.total)
+    amount_paid_before = Payment.compute_order_payments(order)
+    Money.sub!(total_amount, amount_paid_before)
+  end
+
+  defp process_payment_chk(context, multi, payment_method, payment, order, amount_to_pay) do
+    params = %{
+      amount: amount_to_pay,
+      order_id: order.id,
+      slug: "check-payment"
```

Comment:
Umm... there's a unique_constraint on `slug`. Could you change the `Payment` schema to use the new `Ecto.Nanoid`, also rename the `:slug` column to `:number`.